packages/taquito-sapling/src/in-memory-spending-key.ts

```diff
@@ -0,0 +1,91 @@
+import { InvalidSpendingKey } from './error';
+import { InMemoryViewingKey } from './in-memory-viewing-key';
+import toBuffer from 'typedarray-to-buffer';
+import { openSecretBox } from '@stablelib/nacl';
+import * as sapling from '@airgap/sapling-wasm'
+import pbkdf2 from 'pbkdf2';
+import { Prefix, prefix, b58cdecode, b58cencode, validateSpendingKey } from '@taquito/utils';
+import * as bip39 from "bip39";
+
+export class InMemorySpendingKey {
+  #skBuf: Buffer
+  /**
+   *
+   * @param spendingKey unencrypted sask... or encrypted MMXj...
+   * @param password required for MMXj encrypted keys
+   */
+  constructor(spendingKey: string, password?: string) {
+
+    const keyArr = b58cdecode(spendingKey, prefix[Prefix.SASK])
+    // exit first if no password and key is encrypted
+    if (!password && spendingKey.slice(0, 4) !== 'sask') {
+      throw new InvalidSpendingKey(spendingKey)
+    }
+
+    if (password && spendingKey.slice(0, 4) !== 'sask') {
+      const salt = toBuffer(keyArr.slice(0, 8))
+      const encryptedSk = toBuffer(keyArr.slice(8))
+
+      const encryptionKey = pbkdf2.pbkdf2Sync(password, salt, 32768, 32, 'sha512');
+      const decrypted = openSecretBox(
+        new Uint8Array(encryptionKey),
+        new Uint8Array(24),
+        new Uint8Array(encryptedSk),
+        )
+      if (!decrypted) {
+        throw new InvalidSpendingKey(spendingKey)
+      }
+
+      this.#skBuf = toBuffer(decrypted)
+
+    } else {
+    this.#skBuf = toBuffer(keyArr)
+    }
+  }
+
+  /**
+   *
+   * @param mnemonic list of seed words
+   * CHECK
+   * @param derivationPath should be 'm/' check if param needed or if could be hardcoded
```

Comment:
It can be different than `'m/'` based on user preference, so yes, the param is needed. However, it can be set to default to `'m/'` in the method `derivationPath = 'm/'`.

Reply:
👍 great!

Reply:
We can ask other team member opinion as well :)